Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -2559,7 +2576,273 @@ def get_indicator_dbot_object(indicator):
+
+def run_live_response_script_with_polling(client, args):
+    return run_polling_command(client, args, 'microsoft-atp-live-response-run-script', run_live_response_script_action,
+                               get_machine_action_command, get_successfull_action_results_as_info)
+
+
+def run_live_response_script_action(client, args):
+    machine_id = args['machine_id']
+    scriptName = args['scriptName']
+    comment = args['comment']
+    arguments = args.get('arguments')
+    params = [{
+        "key": "ScriptName",
+        "value": scriptName
+    }]
+    if arguments:
+        params.append(
+            {
+                "key": "Args",
+                "value": arguments
+            }
+        )
+    request_body = {
+        "Commands": [
+            {
+                "type": "RunScript",
+                "params": params
+            },
+        ],
+        "Comment": comment
+    }
+
+    # create action:
+    res = client.create_action(machine_id, request_body)
+    return CommandResults(
+        outputs={'action_id': res['id']}
+    )
+
+
+def get_successfull_action_results_as_info(client, res):
+    machine_action_id = res['id']
+    file_link = client.get_live_response_result(machine_action_id, 0)['value']
+
+    f_data = client.download_file(file_link)
+    try:
+        script_result = f_data.json()
+    except Exception:
+        script_result = None
+    return [
+        CommandResults(
+            outputs_prefix='MicrosoftATP.LiveResponseAction',
+            outputs=script_result if script_result else res,
+            readable_output=tableToMarkdown('Script Results:',
+                                            script_result) if script_result else 'Could not retrieve script results.'
+        ),
+        fileResult('Response Result', f_data.content, file_type=EntryType.ENTRY_INFO_FILE)]
+
+
+# -------------- Get File ---------------
+def get_live_response_file_with_polling(client, args):
+    return run_polling_command(client, args, 'microsoft-atp-live-response-get-file', get_live_response_file_action,
+                               get_machine_action_command, get_file_get_successfull_action_results)
+
+
+def get_live_response_file_action(client, args):
+    machine_id = args['machine_id']
+    file_path = args['path']
+    comment = args['comment']
+
+    request_body = {
+        "Commands": [
+            {
+                "type": "GetFile",
+                "params": [{
+                    "key": "Path",
+                    "value": file_path
+                }]
+            },
+        ],
+        "Comment": comment
+    }
+
+    # create action:
+    res = client.create_action(machine_id, request_body)
+    return CommandResults(
+        outputs={'action_id': res['id']}
+    )
+
+
+def get_file_get_successfull_action_results(client, res):
+    machine_action_id = res['id']
+
+    # get file link from action:
+    file_link = client.get_live_response_result(machine_action_id, 0)['value']
```

Comment:
is it possible to not get `value` back?

Reply:
Only if an exception has occurred. In this case I have to raise it anyway in order to show the actual error (action not completed, action already exist, etc)